NAG Build failure: Can't find FLAP

When trying to build `main` with NAG I got:
```
[ 92%] Building Fortran object gridcomps/CMakeFiles/MAPL.gridcomps.dir/MAPL_GridComps.F90.o
NAG Fortran Compiler Release 7.0(Yurakucho) Build 7045
Fatal Error: /Users/mathomp4/Models/MAPL-NAG/MAPL/gridcomps/MAPL_GridComps.F90, line 2: Cannot find module FLAP
             detected at MAPL_CAPMOD@<end-of-statement>
make[2]: *** [gridcomps/CMakeFiles/MAPL.gridcomps.dir/build.make:75: gridcomps/CMakeFiles/MAPL.gridcomps.dir/MAPL_GridComps.F90.o] Error 2
make[1]: *** [CMakeFiles/Makefile2:2431: gridcomps/CMakeFiles/MAPL.gridcomps.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,7 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 ### Changed
 ### Fixed
-- fix format for writing out large number
+
+- Fix format for writing out large number
+- Fix for NAG Build
+
 ## [2.7.1] - 2021-06-11
 
 ### Added

--- a/gridcomps/CMakeLists.txt
+++ b/gridcomps/CMakeLists.txt
@@ -2,6 +2,7 @@ esma_set_this(OVERRIDE MAPL.gridcomps)
 esma_add_library (${this}
      SRCS MAPL_GridComps.F90
      DEPENDENCIES MAPL.base MAPL.pfio MAPL_cfio_r4 MAPL.cap
+                  $<$<BOOL:${BUILD_WITH_FLAP}>:FLAP::FLAP>
      TYPE ${MAPL_LIBRARY_TYPE} 
      )
 

